src/public_tokens/server.rs

```diff
@@ -152,17 +161,31 @@ impl OriginServer {
             *token.token_key_id(),
         );
 
-        let public_key = key_store
-            .get(&truncate_token_key_id(token.token_key_id()))
-            .await
-            .ok_or(RedeemTokenError::KeyIdNotFound)?;
+        let truncated_token_key_id = truncate_token_key_id(token.token_key_id());
+        let public_keys = key_store.get(&truncated_token_key_id).await;
+        if public_keys.is_empty() {
+            return Err(RedeemTokenError::KeyIdNotFound);
+        }
 
         let options = Options::default();
         let signature = Signature(token.authenticator().to_vec());
+        let token_input_bytes = token_input.serialize();
+
+        let mut verified = false;
+        for public_key in public_keys {
+            if signature
+                .verify(&public_key, None, token_input_bytes.clone(), &options)
+                .is_ok()
+            {
+                verified = true;
+                break;
+            }
+        }
```

Comment:
Maybe this is an opportunity to use `iter().any()`? Something like:

```suggestion
        let verified = public_keys.iter().any(|public_key| {
            signature.verify(public_key, None, token_input_bytes.clone(), &options).is_ok()
        });
```

perhaps we can get rid of the `.clone()` on each iteration, too? hmmm